metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -80,29 +79,25 @@ class SemanticdbIndexer(
   private def onChange(path: AbsolutePath, docs: TextDocuments): Unit =
     providers.foreach(_.onChange(docs, path))
 
-  def onChange(file: Path): Unit = {
-    if (!Files.isDirectory(file)) {
-      if (file.isSemanticdb) {
```

Comment:
`if (file.isSemanticdb)` is not needed anymore, we know that we're dealing with semanticdb already